### PR TITLE
#9527: Swapping out most python usages of bcast

### PR DESCRIPTION
**Pull request by eyonland**
### Ticket
#9527 

### Problem description
Removing the dependency on bcast as much as possible so that we have less code to maintain. 
In the process we discovered gaps in how we implemented the broadcasting in that we incorrectly 
handled the repeat operation.   This effort reduces the footprint of the bcast and fixes the broadcast
for interleaved tensors.  When ttnn.repeat supports sharding, bcast should be able to be removed entirely.

### What's changed
Updated a number of places to call ttnn.add, ttnn.multiply, ttnn.subtract instead of ttnn.experimental.bcast

### Checklist
- [x] Post commit CI passes (https://github.com/tenstorrent/tt-metal/actions/runs/10423814469)
- [ ] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (https://github.com/tenstorrent/tt-metal/actions/runs/10292203234)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by cglagovichTT — APPROVED** [review]

**Review by dmakoviichuk-tt — APPROVED** [review]
looks good.
If I understood it right, bcast is the part of any binary op anyway.
But if we need it in python for some other case is it still here right?

**Review by ayerofieiev-tt — APPROVED** [review]

File: docs/source/ttnn/ttnn/dependencies/tt_lib.rst [hunk 1]
```diff
@@ -298,8 +298,6 @@ but in general retaining the data.
 Broadcast and Reduce
 ====================
 
-.. autofunction:: tt_lib.tensor.bcast
```
Comment (ayerofieiev-tt) [inline docs/source/ttnn/ttnn/dependencies/tt_lib.rst:301]: is binding removed?

**Review by arakhmati — APPROVED** [review]

File: models/demos/falcon7b_common/tests/unit_tests/test_falcon_matmuls_and_bmms_with_mixed_precision.py [hunk 2]
```diff
@@ -407,14 +407,11 @@ def test_falcon7b_attnention_sliced(
...
-            ttnn.experimental.tensor.BcastOpMath.MUL,
-            ttnn.experimental.tensor.BcastOpDim.HW,
-            output_mem_config=height_sharded_memory_config,
-            in_place=True,
-        )
+        # This should be done in-place.  This will be done by using an optional output tensor passed to multiply
+        # but this feature is being done on a separate effort.
+        mm_slice_temp = mm_slice
+        mm_slice = ttnn.multiply(mm_slice_temp, reference_scalar, memory_config=height_sharded_memory_config)
+        mm_slice_temp.deallocate()
```
Comment (skhorasganiTT) [inline models/demos/falcon7b_common/tests/unit_tests/test_falcon_matmuls_and_bmms_with_mixed_precision.py:414]: Why does this multiply use a temp copy for the input, while the others don't?

File: docs/source/ttnn/ttnn/dependencies/tt_lib.rst (hunk 1 above)
Comment (eyonland) [reply docs/source/ttnn/ttnn/dependencies/tt_lib.rst:301]: not on this PR (yet)

**Review by tt-rkim — APPROVED** [review]

File: models/demos/falcon7b_common/tests/unit_tests/test_falcon_matmuls_and_bmms_with_mixed_precision.py (hunk 2 above)
Comment (eyonland) [reply models/demos/falcon7b_common/tests/unit_tests/test_falcon_matmuls_and_bmms_with_mixed_precision.py:414]: Fixing this now...

File: models/experimental/bert/fused_ops/linear.py [hunk 3]
```diff
@@ -27,13 +28,11 @@ def Linear(
...
     def linear_(activation):
+        nonlocal bias
         weight_T = ttnn.transpose(weight, -2, -1)
-        output = ttnn.matmul(activation, weight_T)
-
-        if bias is not None:
-            output_plus_bias = tensor.bcast(output, bias, tensor.BcastOpMath.ADD, tensor.BcastOpDim.H)
-            return output_plus_bias
+        if bias is not None and bias.get_layout() != ttnn.TILE_LAYOUT:
+            bias = ttnn.to_layout(bias, ttnn.TILE_LAYOUT)
```
Comment (tt-aho) [inline models/experimental/bert/fused_ops/linear.py:34]: Shouldn't this check/formatting of bias just be done before `linear_`? Then you don't need nonlocal/have this happen during runtime?

**Review by skhorasganiTT — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Review by djordje-tt — APPROVED** [review]

**Comment by eyonland** [comment]
> looks good. If I understood it right, bcast is the part of any binary op anyway. But if we need it in python for some other case is it still here right?

yeah

File: models/experimental/bert/fused_ops/linear.py (hunk 3 above)
Comment (eyonland) [reply models/experimental/bert/fused_ops/linear.py]: By leaving the conversation of the bias within the _linear function it means the bias will be converted lazily and just-in-time.   This means that the calling function could do other things before the invocation.   Maybe I'm overthinking here though.

Comment (tt-aho) [reply models/experimental/bert/fused_ops/linear.py]: I think most fns/classes we wrote for models aren't written in this JIT manner, and does the formatting when the fn/class is initialized. This is also a potentially minor perf hit we are paying for the layout check each time it is called, when we can avoid it by doing it outside during initialization.

**Review by esmalTT — APPROVED** [review]
Please ensure SD test you modified still works on N300 (nightly). Otherwise, LGTM.

File: models/experimental/bert/fused_ops/linear.py (hunk 3 above)
Comment (eyonland) [reply models/experimental/bert/fused_ops/linear.py]: Ok, I will change it

**Review by tt-aho — APPROVED** [review]

File: models/demos/t3000/falcon40b/tt/model_utils.py [hunk 4]
```diff
@@ -480,19 +480,15 @@ def fused_partial_layernorm(
             )
 
             # Apply first layernorm gamma+beta
-            xs_output_slice_1 = ttnn.experimental.tensor.bcast(
+            xs_output_slice_1 = ttnn.multiply(
```
Comment (johanna-rock-tt) [inline models/demos/t3000/falcon40b/tt/model_utils.py:483]: How does performance compare? Do we get ~similar device perf with this change?

**Pull request merged**